api/controllers/game/stalemate.js

```diff
@@ -32,14 +33,16 @@ module.exports = async function (req, res) {
         result: gameService.GameResult.STALEMATE,
       };
       updatePromises.push(gameService.clearGame({ userId }));
-      if (game.isRanked) {
-        updatePromises.push(sails.helpers.addGameToMatch(game));
-      }
     }
 
     updatePromises.push(Game.updateOne({ id: gameId }).set(gameUpdates));
     await Promise.all(updatePromises);
 
+    if (victory.gameOver && gameUpdates.result === gameService.GameResult.STALEMATE && game.isRanked) {
```

Comment:
Why flatten here? Previously this was inside the above conditional i.e. in the case where we are already ending the game, which seems fairly straightfoward to me. Is it about decreasing the total levels of nesting? That's a reasonable concern, idk what to make of it. I think if we do keep this here that we might not need the stalemate check at this moment because it should be the case that the only way the game can end from this endpoint is because of a stalemate
```suggestion
    if (victory.gameOver && game.isRanked) {
```

Reply:
it's the database operation. `await sails.helpers.addGameToMatch(game);` needs to happen after `Promise.all()` otherwise the game state would be `incomplete` even if it is actually `stalemate`

Reply:
Which game state? The one on `game` itself? Is the reason that this is bad because the resulting value of `victory.currentMatch()` is out of date if the game `result` hasn't yet been updated?

Reply:
`victory.currentMatch.games.map(game => game.result)`. `result` would be `incomplete` if we do it the orginal way.

Reply:
That's specifically for the current game that we're adding to the match right now, correct? This is reasonable; my only wondering is whether it makes sense to try to skip a database read by updating the game that gets added to the match in memory without re-querying for it. But this may be best as is. More consistent with the db to just re-query